src/handleAction.js

```diff
@@ -7,19 +7,15 @@ export default function handleAction(type, reducers, defaultState) {
     ? type.toString()
     : type;
 
+  const [nextReducer, throwReducer] = isFunction(reducers)
+    ? [reducers, reducers]
+    : [reducers.next, reducers.throw];
+
   return (state = defaultState, action) => {
     // If action type does not match, return previous state
     if (action.type !== typeValue) return state;
 
-    const handlerKey = action.error === true ? 'throw' : 'next';
-
-    // If function is passed instead of map, use as reducer
-    if (isFunction(reducers)) {
-      reducers.next = reducers.throw = reducers;
-    }
-
-    // Otherwise, assume an action map was passed
-    const reducer = reducers[handlerKey];
+    const reducer = action.error === true ? throwReducer : nextReducer;
 
     return isFunction(reducer)
       ? reducer(state, action)
```

Comment:
Added a patch; you can tell me what you think. We're definitely into a place where my grasp of precise javascript semantics is pretty fuzzy.

As for the breaking change thing– I had the same question about a python library. It turns out that the documentation is really the only source of truth; changing behavior that is undocumented, or errors thrown because of use the library incorrectly or in an undocumented way, isn't considered making a breaking change. I'll leave it up to you guys whether you think it counts in this case.

See also the question I asked here: https://github.com/mojombo/semver/issues/248. Basically, I was worried that adding new arguments to a python function would constitute a breaking change, because people who were calling the function with those arguments in a prior version and seeing a `TypeError` wouldn't see that anymore.